shared/crypto/keypair.go

```diff
@@ -4,13 +4,14 @@ import (
 	"bytes"
 	"crypto/ed25519"
 	"encoding/gob"
+	"encoding/json"
 )
 
 // Encoding is used to serialise the data to store the KeyPairs in the database
 func init() {
 	gob.Register(Ed25519PublicKey{})
 	gob.Register(ed25519.PublicKey{})
-	gob.Register(encKeyPair{})
+	gob.Register(EncKeyPair{})
```

Comment:
Here we should really reduce the scope and just use the `KeyPair` interface. I had the same issue when trying to create an empty one see the function `GetKeypair()` in `keypair.go` this can be used instead of `var kp crypto.KeyPair` when you need it